python-test/features/steps/login.py

```diff
@@ -18,7 +18,40 @@ def check_registered_account(context):
     authenticate(email, password)
 
 
-@when('request referred account registration using registered email, {password_status} password, {username} user name and {company} company name')
+@when("user request account registration {email} email, {password} password, {username} user name and {company} "
+      "company name")
+def check_account_input(context, email, password, username, company):
+    assert_that(email, any_of(equal_to('with'), equal_to('without')),
+                "Not expected option to email")
+    assert_that(password, any_of(equal_to('with'), equal_to('without')),
+                "Not expected option to password")
+    assert_that(username, any_of(equal_to('with'), equal_to('without')),
+                "Not expected option to username")
+    assert_that(company, any_of(equal_to('with'), equal_to('without')),
+                "Not expected option to company")
+    account_input = {'email': None, 'password': None, 'company': None, 'username': None, 'reg_status': 201,
+                     'auth_status': 201}
+    if email == "with":
+        account_input['email'] = f"test_email_{random_string(3)}@email.com"
+    if password == "with":
+        account_input['password'] = configs.get('password')
+    if username == "with":
+        account_input['username'] = f"test_user {random_string(3)}"
+    if company == "with":
+        account_input['company'] = f"test_company {random_string(3)}"
+    if email == "without" or password == "without":
+        account_input['reg_status'] = 400
+        account_input['auth_status'] = 400
+    if email == "with" and password == "without":
+        account_input['auth_status'] = 403
```

Comment:
What if we rewrite this part to something like that ?

`
kvargs = {
    'email': email, 
    'password': password, 
    'username': username, 
    'company': company
}

for key, value in kvargs.items():
   assert_that(value, any_of(equal_to('with'), equal_to('without')),
                f"Not expected option to { key }")
        
account_input = {
    'email': None, 
    'password': None, 
    'company': None, 
    'username': None, 
    'reg_status': 201,
    'auth_status': 201
}

if email == "with":
    account_input['email'] = f"test_email_{random_string(3)}@email.com"
    
    if password == "without":
        account_input['auth_status'] = 403
elif password == "without":
    account_input['reg_status'] = 400
    account_input['auth_status'] = 400
    
if password == "with":
    account_input['password'] = configs.get('password')
    
if username == "with":
    account_input['username'] = f"test_user {random_string(3)}"
    
if company == "with":
    account_input['company'] = f"test_company {random_string(3)}"
`

Reply:
```
elif password == "without":
account_input['reg_status'] = 400
account_input['auth_status'] = 400
```

exactly that way it doesn't work because we have to consider when is without email and with password.. but I adapted your idea..
If you have another suggestion, please share